app/questionnaire/questionnaire_schema.py

```diff
@@ -421,28 +478,11 @@ def _get_error_messages(self):
         return messages
 
 
-def get_nested_schema_objects(parent_object, list_key):
-    """
-    Generic method to extract a flattened list of child objects from a parent
-    object and return an ID-keyed dictionary of those child objects.
-
-    This method also patches on a `parent_id` attribute to each child object.
-
-    :param parent_object: dict containing a list
-    :param list_key: key of the nested list to extract
-    """
-    nested_objects = OrderedDict()
-
-    for parent_id, child_object in parent_object.items():
-        for child_list_object in child_object.get(list_key, []):
-            # patch the ID of the parent onto the object
-            child_list_object["parent_id"] = parent_id
-            nested_objects[child_list_object["id"]] = child_list_object
-
-    return nested_objects
+def is_hashable(value):
+    return hasattr(value, "__hash__") and callable(value.__hash__)
 
 
-def _get_values_for_key(block, key, ignore_keys=None):
+def get_values_for_key(block, key, ignore_keys=None):
```

Comment:
If we are making this a public method it should probably go into schema utils module